conjure-go-client/httpclient/client_test.go

```diff
@@ -29,3 +34,35 @@ func TestNoBaseURIs(t *testing.T) {
 	_, err = client.Do(context.Background(), httpclient.WithRequestMethod("GET"))
 	require.Error(t, err)
 }
+
+func TestMiddlewareCanReadBody(t *testing.T) {
+	unencodedBody := "body"
+	encodedBody, err := codecs.Plain.Marshal(unencodedBody)
+	require.NoError(t, err)
+	bodyIsCorrect := func(body io.ReadCloser) {
+		content, err := ioutil.ReadAll(body)
+		require.NoError(t, err)
+		require.Equal(t, encodedBody, content)
+	}
+
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		bodyIsCorrect(req.Body)
+		rw.WriteHeader(http.StatusOK)
+	}))
+	defer server.Close()
+
+	client, err := httpclient.NewClient(
```

Comment:
What would have failed here if you hadn't made the change? The `req.GetBody()`?

Reply:
yeah the body would have been nil. it's set by the `bodyMiddleware` so rearranging the order of the two allows custom middleware to read the body